src/lib/libdylink.js

```diff
@@ -48,6 +48,62 @@ var LibraryDylink = {
     registerWasmPlugin();
     `,
   $preloadedWasm: {},
+
+  $locateLibraryFromFS__deps: ['$FS', '$PATH'],
+  $locateLibraryFromFS: (filename, searchDirs) => {
+    // Find the library in the filesystem
+    // returns [true, abspath] if found, [false, null] otherwise.
+    if (!(typeof FS.findObject === 'function')) {
+      // FS may not have the findObject method if the filesystem is not
+      // used in the build. Skip the search in this case.
+      return [false, null];
+    }
+    var candidates = [];
+    if (PATH.isAbs(filename)) {
+      candidates.push(filename);
+    } else if (searchDirs) {
+      for (var i = 0; i < searchDirs.length; i++) {
+        candidates.push(PATH.join(searchDirs[i], filename));
```

Comment:
I'm concerned that `PATH.join()` mishandles symlinks, I'll double check later.

Reply:
I haven't thought about symlink behavior in an emscripten environment, could it be an issue with a filesystem like NODEFS?

Reply:
I've dealt with issues like this a lot lately:
* https://github.com/emscripten-core/emscripten/pull/23000
https://github.com/emscripten-core/emscripten/pull/23136
https://github.com/emscripten-core/emscripten/pull/23072
https://github.com/emscripten-core/emscripten/pull/23002
https://github.com/emscripten-core/emscripten/pull/23307

I think the appropriate code here is:
```suggestion
        candidates.push(searchDirs[i] + '/' + filename);
```